bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/base/conditions/evaluation/BasePermissionEmittingService.java

```diff
@@ -113,9 +117,28 @@ private ServiceInvocationResult<Emission> emitFor(Condition condition, Condition
 							condition.evaluationInstructions().expression()), //
 					e);
 		}
+		Optional<Trouble> trouble = checkPeriod(condition);
+		if (trouble.isPresent()) {
+			return new BaseServiceInvocationResult<>(//
+					new BaseDiagnostic(Collections.singletonList(trouble.get())), //
+					new Emission(pack));
+		}
 		return createFor(condition, pack, product);
 	}
 
+	private Optional<Trouble> checkPeriod(Condition condition) {
+		ZonedDateTime now = ZonedDateTime.now();
+		ZonedDateTime start = from(condition.validityPeriod()); // TODO: #566015
+		ZonedDateTime end = expiration(condition.validityPeriod());
+		Optional<TroubleCode> troubles = Optional.empty();
+		if (now.isBefore(start)) {
+			troubles = Optional.of(new LicenseNotStarted());
+		} else if (now.isAfter(end)) {
+			troubles = Optional.of(new LicenseExpired());
+		}
+		return troubles.map(c -> new Trouble(c, c.explanation()));
```

Comment:
Code explanation is static regarding the situation and thus cannot be used to describe an actual trouble at hand. Use dynamic info with actual details that will help a user to define reason of denial.